src/OVAL/probes/unix/sysctl_probe.c

```diff
@@ -268,6 +280,76 @@ int sysctl_probe_main(probe_ctx *ctx, void *probe_arg)
 
         return (0);
 }
+
+#elif defined(OS_FREEBSD)
+int sysctl_probe_main(probe_ctx *ctx, void *probe_arg)
+{
+        FILE *fp;
+        char output[LINE_MAX];
+        const char* SEP = "=";
+        char* mib;
+        char* sysval;
+        SEXP_t *se_mib;
+        SEXP_t *name_entity, *probe_in;
+
+        probe_in    = probe_ctx_getobject(ctx);
+        name_entity = probe_obj_getent(probe_in, "name", 1);
+
+        if (name_entity == NULL) {
+                dE("Missing \"name\" entity in the input object");
+                return (PROBE_ENOENT);
+        }
+
+        fp = popen(SYSCTL_CMD, "r");
+
+        if (!fp) {
+                dE("Failed to open output of %s\n", SYSCTL_CMD);
+		return (PROBE_EFATAL);
+        }
+
+        while (fgets(output, sizeof(output), fp)) {
+		char *strp;
+		mib = strtok_r(output, SEP, &strp);
+		sysval = strtok_r(NULL, SEP, &strp);
+
+		if (!mib)
+			continue;
+
+		if (!sysval)
+			continue;
+
+		se_mib = SEXP_string_new(mib, strlen(mib));
+
+		if (!se_mib) {
+			dE("Failed to allocate new SEXP_string for se_mib");
+			return (PROBE_ENOENT);
+		}
+
+		/* Remove newline */
+		sysval[strlen(sysval)-1] = '\0';
```

Comment:
```suggestion
		sysval[strnlen(sysval, LINE_MAX)-1] = '\0';
```
Or somehow handle the case where `sysval` is not properly terminated.

Reply:
It is my understanding that `strtok` and its variants always return a null-terminated string, perhaps I'm missing something obvious here.

Reply:
Yep, scratch that. As `fgets` is used to fill the buffer, it'd be always terminated properly.